lib/Controller/RoomController.php

```diff
@@ -834,14 +965,26 @@ protected function formatParticipantList(array $participants, bool $includeStatu
 		return new DataResponse(array_values($results), Http::STATUS_OK, $headers);
 	}
 
+	/**
+	 * Add a participant to a room
+	 *
+	 * @param string $newParticipant New participant
+	 * @param string $source Source of the participant
+	 * @return DataResponse<Http::STATUS_OK, array{type: ?int}, array{}>|DataResponse<Http::STATUS_NOT_FOUND|Http::STATUS_NOT_IMPLEMENTED, array<empty>, array{}>|DataResponse<Http::STATUS_BAD_REQUEST, array{error: ?string}, array{}>
+	 * return DataResponse<Http::STATUS_OK|Http::STATUS_BAD_REQUEST|Http::STATUS_NOT_FOUND|Http::STATUS_NOT_IMPLEMENTED, ?array{type?: int, error?: string}, array{}>
+	 *
+	 * 200: Participant successfully added
+	 * 400: Adding participant is not possible
+	 * 404: User, group or other target to invite was not found
+	 */
 	#[NoAdminRequired]
 	#[RequireLoggedInModeratorParticipant]
 	public function addParticipantToRoom(string $newParticipant, string $source = 'users'): DataResponse {
 		if ($this->room->getType() === Room::TYPE_ONE_TO_ONE
 			|| $this->room->getType() === Room::TYPE_ONE_TO_ONE_FORMER
 			|| $this->room->getType() === Room::TYPE_NOTE_TO_SELF
 			|| $this->room->getObjectType() === 'share:password') {
-			return new DataResponse([], Http::STATUS_BAD_REQUEST);
+			return new DataResponse(['error' => null], Http::STATUS_BAD_REQUEST);
```

Comment:
Can we roll this back? empty array should be supported totally fine and since it changes the returned type of "error" it would potentially break GSON and other libs when the client defined it as "its there and string, or absent"

Reply:
Yes, but then you need to add `array<empty>` to the union type

Reply:
Done in https://github.com/nextcloud/spreed/pull/10186/commits/59d25a7abcb951db211b6d70369904403a2d18a8